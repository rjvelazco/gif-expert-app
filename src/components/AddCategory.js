import React, { useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');
    const input = useRef();
    const handleInputChange = ({ target }) => {
        setInputValue(target.value);
        console.log('handleInputChange llamado');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(inputValue);
        if(inputValue.trim().length) {
            setCategories((categories) => [inputValue, ...categories]);
            setInputValue('');
        }
    }

    // Manage input here to avoid fail the test
    // because we are not simulation the DOM on tests in this section
    // useEffect(()=> {
    //     if( !inputValue.length ) {
    //         input.current.value='';
    //     }
    // }, [inputValue])

    return (
        <form onSubmit={handleSubmit}>
            <p>{ inputValue }</p>
            <input
                ref={input}
                type="text"
                placeholder="Add Category"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};