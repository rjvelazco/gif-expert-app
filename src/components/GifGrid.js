import React from 'react'

export const GifGrid = ({ category }) => {
    const api_key = 'GyTgDRvJi4SIjxKQSRDSKA5CNussIOTV';
    const getGifs = async () => {

        const url = `https://api.giphy.com/v1/gifs/search?q=${category}&limit=10&api_key=${api_key}`;
        const resp = await fetch( url);
        const { data } = await resp.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })

        console.log(gifs);

    }
    getGifs();
    return (
        <div>
            <h3>{ category }</h3>
        </div>
    )
}
