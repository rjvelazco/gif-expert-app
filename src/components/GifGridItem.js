import React from 'react'

export const GifGridItem = ({category, title, url}) => {
    const repTitle =  `${category } - gif`.toUpperCase();
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{ title ? title : repTitle }</p>
        </div>
    )
}
