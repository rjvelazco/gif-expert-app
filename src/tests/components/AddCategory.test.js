import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Tests on <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('It should render <AddCategory /> component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('It should change input box', () => {
        const input = wrapper.find('input');
        const value = 'Hello World';
        input.simulate('change', { target: { value } });

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( value );
    });


    test('It should NOT send the information on submit', () => {
        const form = wrapper.find('form');
        // preventDefault(){} => forma corta de preventDefault: () => {};
        form.simulate('submit', {
            preventDefault(){}
        });

        wrapper.update();
        expect( setCategories ).not.toHaveBeenCalled();

    });

    test('It should call the setCategories and clean the input box', () => {

        // 1. simular el inputChange
        wrapper.find('input').simulate('change', { target: { value: 'New Message' } });

        // 2. simular el submit
        wrapper.find('form').simulate('submit', {
            preventDefault(){}
        });

        // 3. setCategories se debe de haber llamado
        expect( setCategories ).toHaveBeenCalled();

        // 4. el valor del input debe estar ''
        expect(  wrapper.find('input').prop('value') ).toBe( '' );

    });

});