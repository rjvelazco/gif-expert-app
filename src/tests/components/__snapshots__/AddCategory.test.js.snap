// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AddCategory /> It should render <AddCategory /> component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Add Category"
    type="text"
    value=""
  />
</form>
`;
